### PR TITLE
fixes issue with generting scaling plans again although they are already available

### PR DESCRIPTION
**Pull request by nyuuyn**
**Fixes**: #294 
However it seems it didn't work only for scale out plans, but thats fixed now. Can't reproduce for other plans.
Tested with the following CSAR: https://drive.google.com/file/d/1V0nwhw_axdsohr83MCnj48JLhfEsYcUe/view?usp=sharing

### CODE REVIEW

File: org.opentosca.planbuilder/org.opentosca.planbuilder.core.bpel/src/main/java/org/opentosca/planbuilder/core/bpel/typebasedplanbuilder/BPELScaleOutProcessBuilder.java [hunk 1]
```diff
@@ -271,8 +273,21 @@ public List<BPELPlan> buildScalingPlans(final Csar csar, final TDefinitions defi
...
             fetchScalingPlansDefinitions(serviceTemplate.getTopologyTemplate(), tagMap, csar);
 
-        for (final ScalingPlanDefinition scalingPlanDefinition : scalingPlanDefinitions) {
+        final List<ScalingPlanDefinition> filteredScalingPlanDefinitions = Lists.newArrayList();
 
+        for (ScalingPlanDefinition scalingPlanDefinition : scalingPlanDefinitions) {
+            boolean isAlreadyAvailable = false;
+            for (TPlan plan : serviceTemplate.getPlans()) {
+                if (plan.getId().endsWith(scalingPlanDefinition.name)) {
+                    isAlreadyAvailable = true;
```
Comment (wederbn) [inline org.opentosca.planbuilder/org.opentosca.planbuilder.core.bpel/src/main/java/org/opentosca/planbuilder/core/bpel/typebasedplanbuilder/BPELScaleOutProcessBuilder.java:282]: break the loop here?

**Review by lharzenetter — CHANGES_REQUESTED** [review]
What @wederbn said.

File: org.opentosca.planbuilder/org.opentosca.planbuilder.core.bpel/src/main/java/org/opentosca/planbuilder/core/bpel/typebasedplanbuilder/BPELScaleOutProcessBuilder.java (hunk 1 above)
Comment (nyuuyn) [reply org.opentosca.planbuilder/org.opentosca.planbuilder.core.bpel/src/main/java/org/opentosca/planbuilder/core/bpel/typebasedplanbuilder/BPELScaleOutProcessBuilder.java]: fixed

**Review by lharzenetter — APPROVED** [review]
LGTM

**Review by wederbn — APPROVED** [review]
LGTM

**Pull request merged**